transformations/compress_whitespace.go

```diff
@@ -4,30 +4,46 @@
 package transformations
 
 import (
-	"unicode"
+	"github.com/corazawaf/coraza/v3/internal/strings"
 )
 
 func compressWhitespace(value string) (string, error) {
-	var a []byte
-	i := 0
-	inWhiteSpace := false
-	length := len(value)
+	for i := 0; i < len(value); i++ {
+		if isLatinSpace(value[i]) {
+			return doCompressWhitespace(value, i), nil
+		}
+	}
+	return value, nil
+}
+
+func doCompressWhitespace(input string, pos int) string {
+	// The output may be significantly different length than input, so we don't preallocate
+	ret := []byte(input[0:pos])
 
-	for i < length {
```

Comment:
Note that previously the iteration was happening per byte, not rune so the use of `unicode.IsSpace` is deceptive, it was still a latin transformation.

ModSec is too

https://github.com/SpiderLabs/ModSecurity/blob/v3/master/src/actions/transformations/compress_whitespace.cc

For this transformation, it feels like a unicode transform could be better but left it matching both modsec and current behavior for now